testing/simapp/app.go

```diff
@@ -892,14 +908,14 @@ func BlockedAddresses() map[string]bool {
 func initParamsKeeper(appCodec codec.BinaryCodec, legacyAmino *codec.LegacyAmino, key, tkey storetypes.StoreKey) paramskeeper.Keeper {
 	paramsKeeper := paramskeeper.NewKeeper(appCodec, legacyAmino, key, tkey)
 
-	paramsKeeper.Subspace(authtypes.ModuleName)
-	paramsKeeper.Subspace(banktypes.ModuleName)
-	paramsKeeper.Subspace(stakingtypes.ModuleName)
-	paramsKeeper.Subspace(minttypes.ModuleName)
-	paramsKeeper.Subspace(distrtypes.ModuleName)
-	paramsKeeper.Subspace(slashingtypes.ModuleName)
+	paramsKeeper.Subspace(authtypes.ModuleName).WithKeyTable(authtypes.ParamKeyTable())
+	paramsKeeper.Subspace(banktypes.ModuleName).WithKeyTable(banktypes.ParamKeyTable())
+	paramsKeeper.Subspace(stakingtypes.ModuleName).WithKeyTable(stakingtypes.ParamKeyTable())
+	paramsKeeper.Subspace(minttypes.ModuleName).WithKeyTable(minttypes.ParamKeyTable())
+	paramsKeeper.Subspace(distrtypes.ModuleName).WithKeyTable(distrtypes.ParamKeyTable())
+	paramsKeeper.Subspace(slashingtypes.ModuleName).WithKeyTable(slashingtypes.ParamKeyTable())
 	paramsKeeper.Subspace(govtypes.ModuleName).WithKeyTable(govv1.ParamKeyTable())
-	paramsKeeper.Subspace(crisistypes.ModuleName)
+	paramsKeeper.Subspace(crisistypes.ModuleName).WithKeyTable(crisistypes.ParamKeyTable())
```

Comment:
Adding `WithKeyTable(<params_key_table>)` for each module here in order to register param key-value types in `x/param` for migration.

This registration was previously done in each `NewKeeper` function of the corresponding module. E.g. 

```go
if !paramSpace.HasKeyTable() {
	paramSpace = paramSpace.WithKeyTable(types.ParamKeyTable())
}
```

The param set pair types need to be registered in order to perform the migrations for modules maintaining their own params, otherwise we fail [here](https://github.com/cosmos/cosmos-sdk/blob/v0.47.0-rc2/x/params/types/subspace.go#L168-L171). Essentially, the key-value attributes are maintained in memory in [this map](https://github.com/cosmos/cosmos-sdk/blob/v0.47.0-rc2/x/params/types/table.go#L15-L17).

Reply:
Nice find. Will this be removable in the future?

Reply:
Yeah we can remove this post v7. We will have to do something similar when we migrate params for ibc-go modules.

Reply:
> Yeah we can remove this post v7.

Should we have an issue for this?

Reply:
Created https://github.com/cosmos/ibc-go/issues/3141